purchase_supplier_approve/models/purchase.py

```diff
@@ -0,0 +1,15 @@
+# Copyright 2019 RGB Consulting - Domantas Sidorenkovas
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models, _
+from odoo.exceptions import ValidationError
+
+
+class PurchaseOrder(models.Model):
+    _inherit = "purchase.order"
+
+    @api.multi
+    def button_confirm(self):
+        if any(rec.partner_id.supplier_approval != 'apd' for rec in self):
+            raise ValidationError(_('You cannot confirm a purchase order if the supplier is not approved.'))
```

Comment:
I seem many lines with length > 79 chars, it should be in new line.
I also wonder why it can pass travis test... my editor whould show error like this,
![image](https://user-images.githubusercontent.com/1973598/65011386-89d23d00-d93d-11e9-811e-b4209e854805.png)